python/service/service.py

```diff
@@ -174,6 +206,7 @@ def c2d_sender_thread(self):
                             )
                             with self.registry_manager_lock:
                                 del self.registry_manager
+                                time.sleep(1)
```

Comment:
is there a reason for this random delay

Reply:
This is in a error-recovery block of code, and the delay seemed prudent to let the old registry_manager instance clean itself up.  I was hitting this pretty consistently at some point and I'm not anymore. Either this recovery code does the job or some other bug was fixed.

---
In reply to: [501950818](https://github.com/Azure/iot-sdk-longhaul/pull/13#discussion_r501950818) [](ancestors = 501950818)